lib/modules/versioning/hermit/index.ts

```diff
@@ -0,0 +1,136 @@
+    super(HermitVersioning.versionRegex);
+  }
+
+  private _isValid(version: string): boolean {
+    return super._parse(version) !== null;
+  }
+
+  protected override _parse(version: string): RegExpVersion | null {
+    const parsed = super._parse(version);
+    if (parsed) {
+      return parsed;
+    }
+    const channelVer = HermitVersioning._getChannel(version);
+
+    const groups = this._config?.exec(channelVer)?.groups;
+
+    if (!groups) {
+      return null;
+    }
+
+    const { major, minor, patch, build, prerelease, compatibility } = groups;
+    const release = [];
+
+    if (major) {
+      release.push(Number.parseInt(major, 10));
+    }
+
+    if (minor) {
+      release.push(Number.parseInt(minor, 10));
+    }
+    if (patch) {
+      release.push(Number.parseInt(patch, 10));
+    }
+    if (build) {
+      release.push(Number.parseInt(build, 10));
+    }
+
+    return {
+      release,
+      prerelease: prerelease,
+      compatibility: compatibility,
+    };
+  }
+
+  private static _isChannel(version: string): boolean {
+    return version.startsWith('@');
+  }
+
+  private static _getChannel(version: string): string {
+    return version.substring(1);
+  }
+
+  override isStable(version: string): boolean {
+    if (this._isValid(version)) {
+      return super.isStable(version);
+    }
+
+    // channel and the rest should be considered unstable version
+    // as channels are changing values
+    return false;
+  }
+
+  override isValid(version: string): boolean {
+    return this._isValid(version) || HermitVersioning._isChannel(version);
+  }
+
+  override isLessThanRange(version: string, range: string): boolean {
+    return this._compare(version, range) < 0;
+  }
+
+  protected override _compare(version: string, other: string): number {
+    if (this._isValid(version) && this._isValid(other)) {
+      return super._compare(version, other);
+    }
+
+    const parsedVersion = this._parse(version);
+    const parsedOther = this._parse(other);
+
+    if (parsedVersion === null || parsedOther === null) {
+      if (parsedVersion === null && parsedOther === null) {
+        return version.localeCompare(other);
+      }
+      return parsedVersion === null ? -1 : 1;
+    }
+
+    const versionReleases = parsedVersion.release;
+    const otherReleases = parsedOther.release;
+
+    const maxLength =
+      versionReleases.length > otherReleases.length
+        ? versionReleases.length
+        : otherReleases.length;
+
+    for (let i = 0; i < maxLength; i++) {
+      const verVal = versionReleases[i];
+      const otherVal = otherReleases[i];
+
+      if (
+        verVal !== undefined &&
+        otherVal !== undefined &&
+        verVal !== otherVal
+      ) {
+        return verVal - otherVal;
+      } else if (verVal === undefined) {
+        return 1;
+      } else if (otherVal === undefined) {
+        return -1;
+      }
+    }
+
+    return 0;
+  }
+
+  override matches(version: string, range: string): boolean {
```

Comment:
what's the issue when removing this?

Reply:
what about the other semver like override functions from regex manager?

Reply:
they seem to be fine from what I can see

Reply:
do you still have any other concern on this, @viceice ?